portal-impl/src/com/liferay/portlet/internal/PortletResponseImpl.java

```diff
@@ -429,17 +433,35 @@ public PortletRequestImpl getPortletRequest() {
 
 	@Override
 	public String getProperty(String key) {
-		throw new UnsupportedOperationException();
+		String[] values = (String[])_headers.get(key);
```

Comment:
@ngriffin7a _headers' type is Map<String, Object>, how do you know it is safe to cast it to String[] ?

Reply:
@shuyangzhou The Javadoc for the new Portlet 3.0 `getProperty(String)` method states:
> This method considers only response properties set or added via setProperty(java.lang.String, java.lang.String) or addProperty(java.lang.String, java.lang.String).
See: https://portals.apache.org/pluto/portlet-3.0-apidocs/javax/portlet/PortletResponse.html#getProperty(java.lang.String)
Our implementation of those methods call through to `public void addHeader(String name, String value)` which stores values as type `String[]`

Reply:
@ngriffin7a I don't think this is right. Underneath we use the same _headers map for all types of headers. If I call this getProperty(String key) with a key maps to a non-string[] values, I will get a class cast exception. I think the javadoc's "considers only" term implies for returning null rather than throwing class cast exception.

Reply:
@shuyangzhou OK I'll fix it and send you another PR.